utils/cache/backend_redis.c

```diff
@@ -99,6 +99,84 @@ static status_t redis_set(redisContext* c, const char* const key, const int key_
   return ret;
 }
 
+static status_t redis_list_push(redisContext* c, const char* const key, const int key_size, const void* const value,
+                                const int value_size, const int timeout) {
+  status_t ret = SC_OK;
+  if (c == NULL || key == NULL || value == NULL) {
+    ta_log_error("%s\n", "SC_CACHE_NULL");
+    return SC_CACHE_NULL;
+  }
+
+  redisReply* reply = NULL;
+  reply = redisCommand(c, "LPUSH %b %b", key, key_size, value, value_size);
+  if (reply->type == REDIS_REPLY_ERROR) {
+    ret = SC_CACHE_FAILED_RESPONSE;
+    ta_log_error("%s\n", "SC_CACHE_FAILED_RESPONSE");
+  }
+
+  freeReplyObject(reply);
+  return ret;
+}
+
+static status_t redis_list_at(redisContext* c, const char* const key, const int index, char* res) {
```

Comment:
We might need to get the `res` buffer size to make sure it is sufficient for the result of the given key. Otherwise, buff overflow might occur.